cmd/kubelet/app/options/options.go

```diff
@@ -550,4 +550,9 @@ Runtime log sanitization may introduce significant computation overhead and ther
 
 	// Graduated experimental flags, kept for backward compatibility
 	fs.BoolVar(&c.KernelMemcgNotification, "experimental-kernel-memcg-notification", c.KernelMemcgNotification, "Use kernelMemcgNotification configuration, this flag will be removed in 1.23.")
+
+	// Memory Manager Flags
+	fs.StringVar(&c.MemoryManagerPolicy, "memory-manager-policy", c.MemoryManagerPolicy, "Memory Manager policy to use. Possible values: 'none', 'static'. Default: 'none'")
+	// TODO: once documentation link is available, replace KEP link with the documentation one.
+	fs.Var(cliflag.NewBracketSeparatedSliceMapStringString(&c.ReservedMemory), "reserved-memory", "A comma separated list of bracket-enclosed configuration for memory manager (e.g. {numa-node=0, type=memory, limit=1Gi}, {numa-node=1, type=memory, limit=1Gi}). The total sum for each memory type should be equal to the sum of kube-reserved, system-reserved and eviction-threshold. See more details under https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1769-memory-manager#reserved-memory-flag")
```

Comment:
Is this a net new format we are adding in this PR?  We would generally not do this with a single flag but instead do it via multiple arguments (if I understand the format).  If I am not understanding the format, please described the allowed syntax.

Reply:
@smarterclayton Hi, many thanks for the review.

The syntax is as follows: 

`--reserved-memory=[{numa-node=int,type=string,limit=string}][,][...]`
* `numa-node` index, e.g. `0`
* `type` of memory:
  * `memory` - conventional memory
  * `hugepages-2Mi` or `hugepages-1Gi` - hugepages 
* `limit` - the amount of reserved memory, e.g. `1Gi`

Example usage:

`--reserved-memory={numa-node=0,type=memory,limit=1Gi},{numa-node=1,type=memory,limit=2Gi}`

Is that fine?

Reply:
Complexity wise this seems harder to understand than the decoupled segments:

```
--reserved-memory={numa-node=0,type=memory,limit=1Gi},{numa-node=1,type=memory,limit=2Gi}
```

```
--reserved-memory <ID>=<TYPE>=<QUANTITY>[, ...]
--reserved-memory 0=memory=1Gi,hugepages-1M=2Gi
--reserved-memory 1=memory=2Gi
```

or with a different first delimiter like `:` (the format described above is fairly close to `-o`)

If this syntax isn't sufficient, I will need more examples to review.

Reply:
will it fine to have something like:
```
--reserved-memory 0:memory=1Gi,hugepages-1M=2Gi
--reserved-memory 1:memory=2Gi
```

Reply:
Are numa nodes discontinuous?  It might be better to just require the client to specify one `--reserved-memory-numa-node` arg for every node, and provide empty strings for the ones that are missing.  Then this just becomes:

```
--reserved-memory-numa-node memory=1Gi,hugepages-1M=2Gi --reserved-memory-numa-node '' --reserved-memory-numa-node memory=2Gi
```

How many numa nodes will the biggest node have?  What are the odds that a kubelet would not have a reservation for each numa node?

Reply:
I saw the machine with 8 NUMA nodes, but it limited by the number of sockets, so I think it can be bigger.
I also can easily imagine when a user will want to allocate reserved memory only on NUMA node 0 and will use other NUMA nodes for some telco workload.

Reply:
I used the format
```
--reserved-memory 0:memory=1Gi,hugepages-1M=2Gi
--reserved-memory 1:memory=2Gi
```
done